docs/iris/src/whatsnew/aggregate_directory.py

```diff
@@ -0,0 +1,183 @@
+
+
+def is_compiled_release(root_directory, release):
+    '''Returns True if the requested release.rst file exists.'''
+    result = False
+    compiled_filename = '{!s}{}'.format(release, EXTENSION)
+    compiled_filepath = os.path.join(root_directory, compiled_filename)
+    if os.path.exists(compiled_filename) and os.path.isfile(compiled_filename):
+        result = True
+    return result
+
+
+def find_release_directory(root_directory, release):
+    '''Returns the matching contribution directory or rasises an exception.'''
+    result = None
+    directory_contents = os.listdir(root_directory)
+    compiled_release = is_compiled_release(root_directory, release)
+    if compiled_release:
+        raise OSError("Specified release is already compiled.")
+    for inode in directory_contents:
+        if os.path.isdir(inode):
+            release_directory = is_release_directory(inode, release)
+            if release_directory:
+                result = os.path.join(root_directory, inode)
+                break
+    if not result:
+        raise OSError("Contribution folder for this release does not exist.")
+    return result
+
+
+def generate_header(release):
+    '''Return a list of text lines that make up a header for the document.'''
+    header_text = []
+    title_template = 'What\'s New in {} {!s}\n'
+    title_line = title_template.format(SOFTWARE_NAME, release)
+    title_underline = ('=' * (len(title_line) - 1)) + '\n'
+    isodatestamp = datetime.date.today().strftime('%Y-%m-%d')
+    header_text.append(title_line)
+    header_text.append(title_underline)
+    header_text.append('\n')
+    header_text.append(':Release: {!s}\n'.format(release))
+    header_text.append(':Date: {}\n'.format(isodatestamp))
+    header_text.append('\n')
+    description_template = 'This document explains the new/changed features '\
+                           'of {} in version {!s}\n'
+    header_text.append(description_template.format(SOFTWARE_NAME, release))
+    header_text.append('(:doc:`View all changes <index>`.)')
+    header_text.append('\n')
+    return header_text
+
+
+def read_directory(directory_path):
+    '''Parse the items in a specified directory and return their metadata.'''
+    directory_contents = os.listdir(directory_path)
+    compilable_files_unsorted = []
+    misnamed_files = []
+    for file_name in directory_contents:
+        file_name_elements = CONTRIBUTION_REGEX.match(file_name)
+        try:
+            category = file_name_elements.group('category')
+            isodate = file_name_elements.group('isodate')
+            summary = file_name_elements.group('summary')
+        except (AttributeError, ValueError) as exception:
+            misnamed_files.append(file_name)
+            continue
+        else:
+            date_of_item = datetime.datetime.strptime(isodate,
+                                                      '%Y-%m-%d').date()
+        compilable_files_unsorted.append({'Category': category,
```

Comment:
Iris doesn't do this - CamelCase is a bit too JavaScript, Python tends to more use underscore_separation in such cases.

Python certainly doesn't borrow from JavaScript. Perish the thought!